src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,8 +1,87 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
 public class WorkWithFile {
+    private List<String> listOfFilesData = new ArrayList<>();
```

Comment:
```suggestion
```

Reply:
Why bring this field inside a method if it is used in several methods?

Reply:
Because you divide logic between methods
Perfectly you only read those values in readFromFile, and then parse it in createReport
That is 2 times

From readDataFromFile you expect something to be returned. You should think about how to divide logic, and think about each part separately.
ReadFromFile - means read from file, nothing more or less